packages/apps/job-launcher/server/src/modules/job/job.service.ts

```diff
@@ -1326,10 +1306,11 @@ export class JobService {
     }
 
     if (jobEntity.requestType === JobRequestType.FORTUNE) {
-      const data = await this.storageService.download(finalResultUrl);
-      const result = typeof data === 'string' ? JSON.parse(data) : data;
+      const data = (await this.storageService.downloadJsonLikeData(
+        finalResultUrl,
+      )) as any;
 
-      if (!result) {
```

Comment:
When file is not found, we get `[]` (both before and after refactoring), so seems like we never get to this condition.

Should we define the type of `data` here as array of fortune results?

Reply:
> When file is not found, we get [] (both before and after refactoring), so seems like we never get to this condition.

Maybe we can check the array's lenght

> Should we define the type of data here as array of fortune results? 

lgtm

Reply:
@portuu3 
Done in https://github.com/humanprotocol/human-protocol/pull/2756/commits/a8bdcf9171a267c573c35e43ca46546164df63d1